lib/src/platform/src/platform.dart

```diff
@@ -51,4 +50,27 @@ class Platform {
       }
     }
   }
+
+  /**
+   * Call a platform method which always provides a result.
+   */
+  static Future<T> invokePlatformMethodNonNull<T>(String method,
```

Comment:
I think refactoring it would be difficult currently. This is because `invokeRequest` wraps the data in an AblyMessage:
```
   final message = (null != argument)
        ? AblyMessage(AblyMessage(argument, handle: _handle))
        : AblyMessage(_handle);
```

I would like to remove the nested AblyMessages, and i have a draft PR for this: https://github.com/ably/ably-flutter/pull/199

Then, we can potentially merge these two method implementations.

Reply:
That makes sense